packages/flutter/lib/src/gestures/force_press.dart

```diff
@@ -0,0 +1,260 @@
+  ///
+  /// The [globalPosition] argument must not be null.
+  ForcePressDetails({
+    this.globalPosition = Offset.zero,
+    this.sourceTimeStamp,
+  }) : assert(globalPosition != null);
+
+  /// The global position at which the function was called.
+  final Offset globalPosition;
+
+  /// Recorded timestamp of the source pointer event that triggered the drag
+  /// event.
+  ///
+  /// Could be null if triggered from proxied events such as accessibility.
+  final Duration sourceTimeStamp;
+}
+
+/// Details object for callbacks that use [GestureForcePressUpdateCallback].
+///
+/// See also:
+///
+///  * [ForcePressGestureRecognizer.onUpdate], which uses [ForcePressUpdateCallback].
+///  * [ForcePressDetails], the details for [GestureForcePressStartCallback],
+///    [GestureForcePressPeakCallback] and [GestureForcePressEndCallback].
+class ForcePressUpdateDetails {
+  /// Creates details for a [GestureForcePressUpdateCallback].
+  ///
+  /// The [globalPosition] argument must not be null.
+  ForcePressUpdateDetails({
+    this.globalPosition = Offset.zero,
+    this.pressure = 0.0,
+    this.sourceTimeStamp,
+  }) : assert(globalPosition != null),
+       assert(pressure != null);
+
+  /// The global position at which the pointer contacted the screen.
+  final Offset globalPosition;
+
+  /// The pressure of the pointer on the screen.
+  final double pressure;
+
+  /// Recorded timestamp of the source pointer event that triggered the drag
+  /// event.
+  ///
+  /// Could be null if triggered from proxied events such as accessibility.
+  final Duration sourceTimeStamp;
+}
+
+/// Signature used by a [ForcePressGestureRecognizer] for when a pointer has
+/// pressed with at least [ForcePressGestureRecognizer.startPressure].
+typedef GestureForcePressStartCallback = void Function(ForcePressDetails details);
+
+/// Signature used by [ForcePressGestureRecognizer] for when a pointer that has
+/// pressed with at least [ForcePressGestureRecognizer.peakPressure].
+typedef GestureForcePressPeakCallback = void Function(ForcePressDetails details);
+
+/// Signature used by [ForcePressGestureRecognizer] during the frames
+/// after the triggering of a [ForcePressGestureRecognizer.onStart] callback.
+typedef GestureForcePressUpdateCallback = void Function(ForcePressUpdateDetails details);
+
+/// Signature for when the pointer that previously triggered a
+/// [ForcePressGestureRecognizer.onStart] callback is no longer in contact
+/// with the screen.
+typedef GestureForcePressEndCallback = void Function(ForcePressDetails details);
+
+/// Recognizes a force press on devices that have force sensors.
+///
+/// Only the force from a single pointer is used to invoke events.
+class ForcePressGestureRecognizer extends OneSequenceGestureRecognizer {
+  /// Creates a force press gesture recognizer.
+  ///
+  /// The [startPressure] defaults to 0.4, and [peakPressure] defaults to 0.85
+  /// where a value of 0.0 is no pressure and a value of 1.0 is maximum pressure.
```

Comment:
So, I don't know if this is the wrong approach, but the events that come in off the device actually have pressures in the range of [0, 6.66], however, I interpolate the recorded values into the range [0,1], so that functionality will work the same across all devices even in this range is different. Maybe it would make more sense to change the default pointer pressure max on these events to 6.66?

Reply:
ya, we're in a bit of a pickle here. Unfortunately neither solution will be very convenient. 

Linearly interpolating is pretty convenient today but the apple API makes no contractual promises with the range or the curve of the values. Neither does Android and things are even more likely to be the wild west there. Trying to shield our users from the rawness of it will likely reach a limit sooner or later. 

Just referring to some arbitrary absolute value would be crappy too since the user has to go capture the device's min/max range first before constructing this recognizer.

I think we have 2 options: 
1- Make getting the device's capability super statically easy (like https://developer.android.com/reference/android/view/InputDevice.MotionRange on Android).
2- Composability is likely our friend. Let there be 2 customizable predicate function arguments where the user can decide whether start and peak are reached given the device's min and max pressure. That function itself can actually have a const default function that maps to a reasonable value but it's overridable. 

@Hixie for opinions and prior art too.

Reply:
We definitely need to address this before merging. 

To @Hixie's comment. We don't really need to worry about subjectively interpreting the 0-1 range. The Android and iOS APIs already contractually assign meanings to those bounds. We can just pass them directly. 

There are no higher bound API definitions. I would prefer letting start and peak be predicates in those cases. The user may want to curve differently if the device can go to 3 vs if it can go to 10. We don't have that future information and shouldn't lock our users to our present interpretation.